compiler/src/dotty/tools/dotc/transform/localopt/StringContextChecker.scala

```diff
@@ -585,21 +584,21 @@ object StringContextMacro {
      *  nothing otherwise
      */
     def checkTypeWithArgs(argument : (Type, Int), conversionChar : Char, partIndex : Int, flags : List[(Char, Int)]) = {
-      val booleans = List(Type.of[Boolean], Type.of[Null])
-      val dates = List(Type.of[Long], Type.of[java.util.Calendar], Type.of[java.util.Date])
-      val floatingPoints = List(Type.of[Double], Type.of[Float], Type.of[java.math.BigDecimal])
-      val integral = List(Type.of[Int], Type.of[Long], Type.of[Short], Type.of[Byte], Type.of[java.math.BigInteger])
-      val character = List(Type.of[Char], Type.of[Byte], Type.of[Short], Type.of[Int])
+      val booleans = List(defn.BooleanType, defn.NullType)
+      val dates = List(defn.LongType, requiredClass("java.util.Calendar").typeRef, requiredClass("java.util.Date").typeRef)
+      val floatingPoints = List(defn.DoubleType, defn.FloatType, requiredClass("java.math.BigDecimal").typeRef)
+      val integral = List(defn.IntType, defn.LongType, defn.ShortType, defn.ByteType, requiredClass("java.math.BigInteger").typeRef)
+      val character = List(defn.CharType, defn.ByteType, defn.ShortType, defn.IntType)
 
       val (argType, argIndex) = argument
       conversionChar match {
         case 'c' | 'C' => checkSubtype(argType, "Char", argIndex, character : _*)
         case 'd' | 'o' | 'x' | 'X' => {
           checkSubtype(argType, "Int", argIndex, integral : _*)
           if (conversionChar != 'd') {
-            val notAllowedFlagOnCondition = List(('+', !(argType <:< Type.of[java.math.BigInteger]), "only use '+' for BigInt conversions to o, x, X"),
-            (' ', !(argType <:< Type.of[java.math.BigInteger]), "only use ' ' for BigInt conversions to o, x, X"),
-            ('(', !(argType <:< Type.of[java.math.BigInteger]), "only use '(' for BigInt conversions to o, x, X"),
+            val notAllowedFlagOnCondition = List(('+', !(argType <:< requiredClass("java.math.BigInteger").typeRef), "only use '+' for BigInt conversions to o, x, X"),
+            (' ', !(argType <:< requiredClass("java.math.BigInteger").typeRef), "only use ' ' for BigInt conversions to o, x, X"),
```

Comment:
All these requiredClass should go in Definitions.scala

Reply:
I still need to do an optimization pass on the ported code.